igcollect/redis.py

```diff
@@ -18,9 +18,14 @@ def parse_args():
 def main():
     args = parse_args()
     cfg = get_redis_conf('requirepass', 'port')
-    redis_info = check_output([
-        'redis-cli', '-a', cfg['requirepass'], '-p', cfg['port'], 'info',
-    ])
+
+    cli_command = ['redis-cli']
+    if 'requirepass' in cfg:
+        cli_command.extend(['-a', cfg['requirepass']])
+    if 'port' in cfg:
+        cli_command.extend(['-p', cfg['port']])
+    cli_command.append('info')
```

Comment:
Would be nicer to have it in an own method.  But not a requirement for being merged.